lib/ts/recipe/thirdparty/types.ts

```diff
@@ -38,39 +38,46 @@ export type UserInfoMap = {
     };
 };
 
-export type ProviderConfigForClientType = {
-    name: string;
-
+export type ProviderClientConfig = {
+    clientType?: string;
     clientID: string;
     clientSecret?: string;
-    scope: string[];
+    scope?: string[];
     forcePKCE?: boolean;
-    additionalConfig?: any;
+    additionalConfig?: { [key: string]: any };
+};
+
+type CommonProviderConfig = {
+    thirdPartyId: string;
+    tenantId?: string;
+    name?: string;
 
     authorizationEndpoint?: string;
-    authorizationEndpointQueryParams?: any;
+    authorizationEndpointQueryParams?: { [key: string]: string };
     tokenEndpoint?: string;
-    tokenEndpointBodyParams?: any;
+    tokenEndpointBodyParams?: { [key: string]: string };
     userInfoEndpoint?: string;
-    userInfoEndpointQueryParams?: any;
-    userInfoEndpointHeaders?: any;
+    userInfoEndpointQueryParams?: { [key: string]: string };
+    userInfoEndpointHeaders?: { [key: string]: string };
     jwksURI?: string;
     oidcDiscoveryEndpoint?: string;
     userInfoMap?: UserInfoMap;
     validateIdTokenPayload?: (input: {
-        idTokenPayload: any;
+        idTokenPayload: { [key: string]: any };
         clientConfig: ProviderConfigForClientType;
+        userContext?: any;
```

Comment:
should not be optional since the user is never going to call this function. They are only ever going to implement the body of this function

Reply:
Done